### ISSUE TITLE
framework not found FirebaseDatabase

### ISSUE THREAD

**Issue by visdyn** [firebase/firebase-ios-sdk#9492]
* Xcode version: __13.3___
  * Firebase SDK version: __8.13.0___
  * Installation method: `CocoaPods` (select one)
  * Firebase Component: __Firestore___
  * Target platform(s): `macOS`

### [REQUIRED] Step 2: Describe the problem

I was using Firestore 8.13.0 and wanted to ensure my pods were all updated so I command-lined `pod update`. Cocoapods then *downgraded* all of my pods to pre-7.0 release to 6.34.0.

After finding issue [7195](https://github.com/firebase/firebase-ios-sdk/issues/7195) and updating my podfile, deintegrating and the `pod install` all of the current versions were installed.

```
Installing Firebase 8.13.0 (was 6.34.0)
Installing FirebaseAuth 8.13.0 (was 6.9.2)
Installing FirebaseCore 8.13.0 (was 6.10.4)
Installing FirebaseCoreDiagnostics 8.13.0 (was 1.7.0)
Installing FirebaseFirestore 8.13.0 (was 1.19.0)
Installing FirebaseFirestoreSwift 8.13.0-beta (was 0.4.0)
Installing GoogleDataTransport 9.1.2 (was 7.5.1)
Installing GoogleUtilities 7.7.0 (was 6.7.2)
Installing PromisesObjC 2.0.0 (was 1.2.12)
Installing abseil 0.20200225.0
Installing nanopb 2.30908.0 (was 1.30906.0)
```

However, the project (it's about 10 lines of code) will no longer compile. The error is

`framework not found FirebaseDatabase`

#### Steps to reproduce:

hmmm... pod install and then build?

#### Relevant Code:

The podfile
```
project 'FirestoreTest.xcodeproj'

platform :osx, '10.15'

target 'FirestoreTest' do
  use_frameworks! :linkage => :static
  
  pod 'Firebase/Auth'
  pod 'Firebase/Firestore'
  pod 'FirebaseFirestoreSwift', '~> 8.0-beta'
end
```

Top of the project

```
import Cocoa
import Firebase
import FirebaseFirestore
import FirebaseFirestoreSwift

class ViewController: NSViewController {
```

The error

<img width="572" alt="framework not found FirebaseDatabase" src="https://user-images.githubusercontent.com/19558530/159352906-9ec3b28b-9a65-4a34-bd2e-178cced2e148.png">

**Comment by paulb777** [comment]
Looks like there is a reference to `FirebaseDatabase` in the Xcode project. `pod deintegrate` and another `pod install` might solve it. Otherwise you'll need to investigate where it's occurring in the project and remove it.  (or add `pod 'FirebaseDatabase'` to the Podfile.

**Comment by visdyn** [comment]
Thanks @paulb777 

No reference to FirebaseDatabase is in the project code. The only reference to it is in Google Firebase.h from the FirebaseCore Pod as shown in the below image, but I am not importing-ing anything like that.

Should the podfile include FirebaseDatabase? e.g. `pod 'Firebase/Database'` even though we are not using the Realtime Database at all?

I added that, deintegrated, `pod install` and then got a different error `framework no found FirebaseStorage`

So, I then added `pod 'Firebase/Storage'`, deintegrated, `pod install` and it now seems to be working.

We are not using the Realtime Database nor Storage in any way so I am not sure what's causing the issue.

Here's my updated podfile which is working at this time.

```
project 'FirestoreTest.xcodeproj'

platform :osx, '10.15'

target 'FirestoreTest' do
  use_frameworks! :linkage => :static
  
  pod 'Firebase/Auth'
  pod 'Firebase/Database'
  pod 'Firebase/Firestore'
  pod 'Firebase/Storage'
  pod 'FirebaseFirestoreSwift', '~> 8.0-beta'
end
```

<img width="709" alt="Firebase Pod" src="https://user-images.githubusercontent.com/19558530/159358803-0bf44828-1a0c-4eda-9591-e07d2d347990.png">

**Comment by paulb777** [comment]
Weird. Try searching the `project.pbx` file in the Xcode project directory in a text editor or grep.

**Comment by visdyn** [comment]
Hmm. I don't have a `project.pbx` file anywhere on my drive.

If you were referring to the `project.pbxproj` file that's inside the XCode `app_name.xcodeproj` package, I opened that and no references to FirebaseDatabase exist

I finally gave up and reconstructed the project from scratch using the podfile I outlined above, which is working correctly.. I will circle back around if I can find a way to consistently duplicate the issue.
